src/services/tx/txSender.ts

```diff
@@ -189,6 +189,23 @@ export const dispatchTxSigning = async (
   return signedTx
 }
 
+/**
+ * On-Chain sign a transaction
+ */
+export const dispatchOnChainSigning = async (safeTx: SafeTransaction) => {
+  const sdk = getAndValidateSafeSDK()
+
+  try {
+    const safeTxHash = await sdk.getTransactionHash(safeTx)
+    await sdk.approveTransactionHash(safeTxHash)
+  } catch (err) {
+    txDispatch(TxEvent.SIGN_FAILED, { txId: undefined, error: err as Error })
+    throw err
+  }
+
```

Comment:
We should dispatch a success event here.

Reply:
Dispatching the SUCCESS event makes sense only after the transaction has been executed. In this case we don't execute or sign the transaction. It is only being proposed to the gateway. I would suggest we only dispatch a FAILED event in case the transaction is being rejected in the wallet.

Reply:
I also added an EXECUTING event dispatch and use a `groupKey` since we don't have a `txId` at that point.